src/slurm_plugin/resume.py

```diff
@@ -177,12 +193,18 @@ def _resume(arg_nodes, resume_config):
     )
     success_nodes = [node for node in node_list if node not in instance_manager.failed_nodes]
     log.info("Successfully launched nodes %s", print_with_count(success_nodes))
-    if instance_manager.failed_nodes:
+
+    ice_nodes = [node for node_list in instance_manager.ice_nodes.values() for node in node_list]
+    if ice_nodes:
+        _handle_ice_nodes(instance_manager)
+
+    failed_not_ice_nodes = [node for node in instance_manager.failed_nodes if node not in ice_nodes]
+    if failed_not_ice_nodes:
         log.error(
             "Failed to launch following nodes, setting nodes to down: %s",
             print_with_count(instance_manager.failed_nodes),
         )
-        _handle_failed_nodes(instance_manager.failed_nodes)
+        _handle_failed_nodes(failed_not_ice_nodes)
```

Comment:
This is a good idea to help us set more specified node reason when there's ClientError. I didn't implement it because it require more code changes when failed_nodes is used both in resume and clustermgtd. Now I changed it to: In case of ClientError we always set the code and failed_nodes into the dict containing error->nodes mapping